OSDOCS-4653: Initial release notes file MicroShift (#54515)
openshift/openshift-docs: microshift_release_notes/microshift-4-12-release-notes.adoc

[id="microshift-4-12-about-this-release"]
== About this release

{product-title} is now available as a Developer Preview software. Features and known issues that pertain to {product-title} {product-version} are included in this topic. For more information about the support scope of Red Hat Developer Preview software, see link:https://access.redhat.com/support/offerings/devpreview/[Developer Preview Support Scope].

[IMPORTANT]
====
Red Hat will not provide or support an update or upgrade path from Developer Preview and Technology Preview versions to later versions of {product-title}. A new installation will be necessary.
====

[id="microshift-known-issues"]
== Known issues
* There is an issue exporting services via the Route API when the host and routerCanonicalHostname do not include the sub domain name. To work around this problem, use the subdomain setting when creating a Route.
+
.Example
[source, terminal]
----
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  name: route-1
spec:
  subdomain: svc-subdomain  ⇐ specify the subdomain
  port:
    targetPort: 8080
  to:
    kind: Service
    name: svc-1
----

> * There is an issue resuming pods when a service is idled. Unidling is not supported by the {product-title} software, and manual re-scaling the replicas is required.

Reviewer: Unidling [isn't actually a word](https://www.merriam-webster.com/dictionary/unidle), is it? I can't think of a good alternative, though.

Author: "Returning a service to an active state" perhaps?

Reviewer 2: We use the term "unidling" elsewhere in the docs. https://access.redhat.com/documentation/en-us/openshift_container_platform/4.1/html/applications/idling-applications#idle-unidling-applications_idling-applications

We should make sure it's clear here that services are not scaled back up automatically.

```suggestion
* There is an issue resuming pods when a service is idled. {product-title} does not support automatically reactivating services, and manual re-scaling the replicas is required.
```

Author: Thank you, @mburke5678 and @dhellmann, edit made.

> * It is possible to over-provision the storage system by creating persistent volumes (PVs) that do not actually fit on the disk when using the `spare-gb` configuration value in `lvmd.conf`. To protect capacity from being used by the `topoLVM` driver, use a new separate volume group (VG) that is not exposed to topolvm.

Reviewer: s/exposed to topolvm/exposed to the `topoLVM` driver ?? Or just `topoLVM`

Reviewer 2: topoLVM is the default CSI driver. Saying just "topoLVM" makes sense in this context.

> * There is a known issue involving `runc` and `systemd`. `Runc` is overly verbose about `mount ops`. These messages are innocuous and do not indicate a problem with the system's operation.  This bug affects any Linux platform running a container runtime built on `runc`.

Reviewer: I think `runc` is always lower-case `r`. Even at the start of the sentence. See https://github.com/opencontainers/runc

+
You can suppress all `runc` mount logs through `systemd`. Create the file `/etc/systemd/system/run-containerd-.mount.d/10-silence.conf`, or whatever prefix you need to silence, with:
+
[source, terminal]
----
[Mount] LogLevelMax=notice
----
+
And reload the daemon.

> * {product-title} uses long-lived certificates but some of the code imported from {ocp} prefers longer lifetimes and logs warning messages. The messages are innocuous and do not indicate a problem with the system's operation.

Reviewer: s/long-lived certificates but some/long-lived certificates. However, some

s/prefers/requires  (I think this qualifies as an [anthropomorphism](https://stylepedia.net/style/6.0/#Avoiding_Slang_Metaphors-MisleadingLanguage-Anthropomorphism), which I stumbled upon earlier today)

Reviewer 2: I missed this one. We fixed it, so the note can come out. cc @pacevedom

Author: @mburke5678 You are right about that, though. So noted.

> * There is an issue with the way {product-title}'s CNI driver manages the firewall rules that means if the firewall settings are modified after {product-title} is started then {product-title} must be restarted. The recommended workaround is to configure firewall rules before starting {product-title}.

Reviewer: Something about _that means_ bugs me. Maybe _such that, if the firewall settings are modified..._

[id="microshift-4-12-asynchronous-errata-updates"]
== Asynchronous errata updates

> Security, bug fix, and enhancement updates for {product-title} {product-version} are released as asynchronous errata through the Red Hat Network. All {product-title} {product-version} errata is https://access.redhat.com/downloads/content/290/[available on the Red Hat Customer Portal]. See the https://access.redhat.com/support/policy/updates/openshift[{product-title} Life Cycle] for more information about asynchronous errata.

Author: You are right, it's _erratum_ in singular. That is straight from OCP docs, unedited. Want to make a bug to fix that there as well? Fixed here.

====
Red Hat will not provide or support an update or upgrade path from Developer Preview and Technology Preview versions to later versions of {product-title}. A new installation will be necessary.
====

[id="microshift-known-issues"]
== Known issues
* There is an issue exporting services via the Route API when the host and routerCanonicalHostname do not include the sub domain name. To work around this problem, use the subdomain setting when creating a Route.
+
.Example
[source, terminal]
----
apiVersion: route.openshift.io/v1
kind: Route
metadata:
  name: route-1
spec:
  subdomain: svc-subdomain  ⇐ specify the subdomain
  port:
    targetPort: 8080
  to:
    kind: Service
    name: svc-1
----

* There is an issue resuming pods when a service is idled. Unidling is not supported by the {product-title} software, and manual re-scaling the replicas is required.

* It is possible to over-provision the storage system by creating persistent volumes (PVs) that do not actually fit on the disk when using the `spare-gb` configuration value in `lvmd.conf`. To protect capacity from being used by the `topoLVM` driver, use a new separate volume group (VG) that is not exposed to topolvm.

> * There is a known issue involving `runc` and `systemd`. `Runc` is overly verbose about `mount ops` and `systemd` does not have the capability to filter out the messages. These messages are innocuous and do not indicate a problem with the system's operation.  This bug affects any Linux platform running a container runtime built on `runc`.

Reviewer 2: ```suggestion
* There is a known issue involving `runc` and `systemd`. `Runc` is overly verbose about `mount ops`. These messages are innocuous and do not indicate a problem with the system's operation.  This bug affects any Linux platform running a container runtime built on `runc`.
```

The next paragraph demonstrates how to set up a filter.

+
You can suppress all `runc` mount logs through `systemd`. Create the file `/etc/systemd/system/run-containerd-.mount.d/10-silence.conf`, or whatever prefix you need to silence, with:
+
[source, terminal]
----
[Mount] LogLevelMax=notice
----
+
And reload the daemon.

* {product-title} uses long-lived certificates but some of the code imported from {ocp} prefers longer lifetimes and logs warning messages. The messages are innocuous and do not indicate a problem with the system's operation.

* There is an issue with the way {product-title}'s CNI driver manages the firewall rules that means if the firewall settings are modified after {product-title} is started then {product-title} must be restarted. The recommended workaround is to configure firewall rules before starting {product-title}.

> * There is a known incompatibility in the Route API in {product-title} when compared with {ocp}. In {product-title}, the default value for the `spec.to.kind` and `spec.wildcardPolicy` fields are only applied if the API request does not include the field at all; an empty string is not replaced by the default value. The workaround is to ensure that all Routes have `spec.to.kind` and `spec.wildcardPolicy` set explicitly.

Reviewer 2: I think we fixed this, didn't we @benluddy ?
